### PR TITLE
oneDNN: work around Sphinx type lookup bug

### PR DESCRIPTION
**Pull request by rsdubtso**
https://github.com/sphinx-doc/sphinx/issues/7740

### CODE REVIEW

File: source/elements/oneDNN/include/dnnl.hpp [hunk 1]
```diff
@@ -1696,24 +1694,24 @@ struct reorder : public primitive {
...
     ///
-    /// @param stream Stream object. The stream must belong to the same engine
+    /// @param astream Stream object. The stream must belong to the same engine
     ///               as the primitive.
     /// @param src Source memory object.
     /// @param dst Destination memory object.
     /// @param deps Vector of SYCL events that the execution should depend on.
     ///
     /// @returns SYCL event that corresponds to the SYCL queue underlying the
     ///          @p stream.
```
Comment (emfomenk) [inline source/elements/oneDNN/include/dnnl.hpp:1712]:
```suggestion
    ///          @p astream.
```

File: source/elements/oneDNN/include/dnnl.hpp [hunk 2]
```diff
@@ -541,17 +541,15 @@ struct engine {
 /// @{
 
 /// A container for stream attributes.
-struct stream_attr : public handle<dnnl_stream_attr_t> {
-    using handle::handle;
-
+struct stream_attr {
```
Comment (emfomenk) [inline source/elements/oneDNN/include/dnnl.hpp:544]: why?

Comment (rsdubtso) [reply source/elements/oneDNN/include/dnnl.hpp:544]: This header file is not supposed to have any mention of `handle<T>` and, especially, should not have any references to the C API, as this is implementation-specific.

File: source/elements/oneDNN/include/dnnl.hpp (hunk 1 above)
Comment (rsdubtso) [reply source/elements/oneDNN/include/dnnl.hpp]: Fixed. Thanks!

File: source/elements/oneDNN/include/dnnl.hpp (hunk 2 above)
Comment (emfomenk) [reply source/elements/oneDNN/include/dnnl.hpp:544]: oh, my bad, i didn't notice we don't have `handle<T>` in other places. sorry for the noise.

**Pull request merged**